### PR TITLE
[FormRecognizer] Applying the IgnoreServiceError property

### LINKED ISSUE #26956: [Form Recognizer] AdminOps(False) failing intermittently in nightly runs
Form Recognizer nightly test runs are failing with:

> Error message
> Expected: not null
> But was:  null
> 
> 
> Stack trace
>    at Azure.AI.FormRecognizer.DocumentAnalysis.Tests.DocumentModelAdministrationClientLiveTests.AdminOps(Boolean useTokenCredential) in /mnt/vss/_work/1/s/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentModelAdministrationClient/DocumentModelAdministrationClientLiveTests.cs:line 193
>    at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.BlockUntilCompleted()
>    at NUnit.Framework.Internal.MessagePumpStrategy.NoMessagePumpStrategy.WaitForCompletion(AwaitAdapter awaiter)
>    at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
>    at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
>    at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
>    at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.<>c__DisplayClass1_0.<Execute>b__0()
>    at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)

For more details check here:

- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1340362&view=results

@jsquire for notification.

### LINKED ISSUE #27498: [Form Recognizer] ManageModelsAsync failing in nightly runs
Form Recognizer nightly test runs are failing with:

> Error message
> Azure.RequestFailedException : Invalid request.
> Status: 400 (Bad Request)
> ErrorCode: InvalidRequest
> 
> Content:
> {
> "error": {
> "code": "InvalidRequest",
> "message": "Invalid request.",
> "innererror": {
> "code": "ContentSourceNotAccessible",
> "message": "Content is not accessible: Invalid data URL"
> }
> }
> }
> 
> Headers:
> Transfer-Encoding: chunked
> x-envoy-upstream-service-time: 60379
> apim-request-id: e121459a-90b4-4051-944c-f20dc5d41ab5
> Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
> X-Content-Type-Options: nosniff
> Date: Fri, 11 Mar 2022 04:54:39 GMT
> Content-Type: application/json
> 
> 
> Stack trace
>    at Azure.AI.FormRecognizer.DocumentAnalysis.DocumentAnalysisRestClient.BuildDocumentModelAsync(BuildDocumentModelRequest buildRequest, CancellationToken cancellationToken) in /Users/runner/work/1/s/sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/DocumentAnalysisRestClient.cs:line 340
>    at Azure.AI.FormRecognizer.DocumentAnalysis.DocumentModelAdministrationClient.StartBuildModelAsync(Uri trainingFilesUri, DocumentBuildMode buildMode, String modelId, BuildModelOptions buildModelOptions, CancellationToken cancellationToken) in /Users/runner/work/1/s/sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentModelAdministrationClient.cs:line 249
>    at Azure.AI.FormRecognizer.DocumentAnalysis.Samples.DocumentAnalysisSamples.ManageModelsAsync() in /Users/runner/work/1/s/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample_ManageModelsAsync.cs:line 50
>    at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.BlockUntilCompleted()
>    at NUnit.Framework.Internal.MessagePumpStrategy.NoMessagePumpStrategy.WaitForCompletion(AwaitAdapter awaiter)
>    at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
>    at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
>    at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
>    at NUnit.Framework.Internal.Execution.SimpleWorkItem.<>c__DisplayClass4_0.<PerformWork>b__0()
>    at NUnit.Framework.Internal.ContextUtils.<>c__DisplayClass1_0`1.<DoIsolated>b__0(Object _)
>    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
>    at NUnit.Framework.Internal.ContextUtils.DoIsolated(ContextCallback callback, Object state)
>    at NUnit.Framework.Internal.ContextUtils.DoIsolated[T](Func`1 func)
>    at NUnit.Framework.Internal.Execution.SimpleWorkItem.PerformWork()

For more details check here:

- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1426426&view=results

@jsquire for notification.

### PR DESCRIPTION
**Pull request by kinelski**
This PR makes our tests ignore a couple of service-side errors we have been facing for months. GitHub issues have been created to track those.

Fixes #26956.
Fixes #27498.

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentModelAdministrationClient/DocumentModelAdministrationClientLiveTests.cs [hunk 1]
```diff
@@ -202,6 +202,7 @@ public async Task GetPrebuiltModel()
         [RecordedTest]
         [TestCase(true)]
         [TestCase(false)]
+        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/28705")]
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentModelAdministrationClient/DocumentModelAdministrationClientLiveTests.cs:205]: We sometimes get an empty list when listing models. Not sure if an issue on the service or client side yet.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/RecognizeBusinessCardsLiveTests.cs [hunk 2]
```diff
@@ -344,7 +344,7 @@ public void StartRecognizeBusinessCardsThrowsForDamagedFile()
         [RecordedTest]
         public void StartRecognizeBusinessCardsFromUriThrowsForNonExistingContent()
         {
-            var client = CreateFormRecognizerClient();
+            var client = CreateFormRecognizerClient(retryFlakyErrors: false);
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/RecognizeBusinessCardsLiveTests.cs:347]: In this test, we are reproducing one of the errors our policy retries. For this reason, I created the `retryFlakyErrors` parameter to tell the method whether we want to retry error or not.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentModelAdministrationClient/DocumentModelAdministrationClientLiveTests.cs [hunk 3]
```diff
@@ -236,8 +238,7 @@ public async Task AdminOps(bool useTokenCredential)
 
             await client.DeleteModelAsync(modelId);
 
-            RequestFailedException ex = Assert.ThrowsAsync<RequestFailedException>(() => client.GetModelAsync(modelId));
-            Console.WriteLine(ex.ErrorCode);
+            Assert.ThrowsAsync<RequestFailedException>(() => client.GetModelAsync(modelId));
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentModelAdministrationClient/DocumentModelAdministrationClientLiveTests.cs:241]: Not sure why we were printing the error code in this test (debugging?).

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/RecognizeCustomFormsLiveTests.cs [hunk 4]
```diff
@@ -231,6 +232,7 @@ public async Task StartRecognizeCustomFormsWithLabelsCanParseMultipageForm(bool
         }
 
         [RecordedTest]
+        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/28556")]
         public async Task StartRecognizeCustomFormsWithLabelsCanParseBlankPage()
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/RecognizeCustomFormsLiveTests.cs:236]: Blank pages have been making the service fail for some time now. Still need to investigate what errors we actually get.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentAnalysisClient/DocumentAnalysisClientLiveTests.cs [hunk 5]
```diff
@@ -20,6 +20,7 @@ namespace Azure.AI.FormRecognizer.DocumentAnalysis.Tests
     /// These tests have a dependency on live Azure services and may incur costs for the associated
     /// Azure subscription.
     /// </remarks>
+    [IgnoreServiceError(400, "InvalidRequest", Message = "Content is not accessible: Invalid data URL", Reason = "https://github.com/Azure/azure-sdk-for-net/issues/28923")]
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentAnalysisClient/DocumentAnalysisClientLiveTests.cs:23]: The `IgnoreServiceError` attribute will make sure we don't fail when we encounter these types of errors. **The test won't be retried**, so this is still not ideal. I created GitHub issues (linked in the `Reason` properties) to keep track of removing these attributes once the service has fixed the issue on their side.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentModelAdministrationClient/DocumentModelAdministrationClientLiveTests.cs (hunk 3 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentModelAdministrationClient/DocumentModelAdministrationClientLiveTests.cs:241]: Not needed for our SDK testing, but at some point we wanted to verify the error codes the service returns as they kept changing them.
They have been stable for a while so we can keep deleting printing/verifications

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/RecognizeCustomFormsLiveTests.cs (hunk 4 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/RecognizeCustomFormsLiveTests.cs:236]: I wonder how much investigation needs to happen in our side if this code hasn't touched and the service also stop changes in v2.1

**Review by maririos — APPROVED** [review]

**Pull request merged**